Add Database Models

Wambita opened the pull request:
This pull request introduces:  
- A **Go data model** (`models/models.go`) defining core entities with relationships.  

### **Changes Made**  
1. **Data Models (`models/models.go`)**  
   - Uses **UUIDs** as unique identifiers.  
   - Implements **struct relationships** for seamless querying.  
   - Includes timestamps (`created_at`, `updated_at`, `deleted_at`) for tracking record changes.  

### **Review Request**  
Looking for feedback on:  
✅ Go struct relationships—Are they well-structured and flexible?  
✅ Do the structs and schema match?


### **Next Steps**  
- Implement database migrations.  
- Develop query functions for CRUD operations.  

Would appreciate any suggestions before merging

Wambita committed:
Feat: add user struct data model

Wambita committed:
Feat: add post struct

Wambita committed:
Feat: add post struct

Wambita committed:
Feat: add post struct

Wambita committed:
Feat: add comment  struct

Wambita committed:
Feat: add comment  struct

Wambita committed:
Feat: add session struct

Wambita committed:
Fix: remove json from the data models

Wambita committed:
Feat: add db schema

Wambita committed:
Merge branch 'main' into models

Wambita committed:
Feat: add database struct

Wambita committed:
Merge branch 'models' of https://github.com/nguonodave/forum into models

Wambita committed:
Fix: remove schema file

WycliffeAlphus reviewed (commented):
Do we really need the creation time in the categories table?

rayjonesjay reviewed (approved):
The `vote` structure should have a dislike and like count and they should be uint32. This is a suggestion.

WycliffeAlphus commented:
@Wambita I am requesting the addition of a sessions table for the session management. The format should be :

CREATE TABLE sessions (
    token TEXT PRIMARY KEY,
    user_id INTEGER NOT NULL,
    expires_at DATETIME NOT NULL,
    FOREIGN KEY (user_id) REFERENCES users(id)
);

Wambita commented:
> @Wambita I am requesting the addition of a sessions table for the session management. The format should be :
> 
> CREATE TABLE sessions ( token TEXT PRIMARY KEY, user_id INTEGER NOT NULL, expires_at DATETIME NOT NULL, FOREIGN KEY (user_id) REFERENCES users(id) );

 @nguonodave  is implementng the schema

nguonodave committed:
refactor(models): adjust models to align with the created schema

- Removed CategoryId in Post and PostId in category tables since
they have a many-to-many relationship
- Removed DeletedAt from Post, soft deleting is not necessary in this scope
- Removed CreatedAt from Category
- Removed ParentId from Comment to avoid nested comments
- Added Id and CommentId in Votes to store the unique id and related comment id
- Change Value in Vote to Type of string to store 'likes' and 'dislikes'

nguonodave committed:
Merge branch 'main' into models

nguonodave committed:
chore(format): format with gofmt
